ca/server.go

```diff
@@ -32,6 +34,45 @@ type APISecurityConfigUpdater interface {
 
 var _ APISecurityConfigUpdater = &Server{}
 
+// ExternalCATLSUpdater is a way to notify the server of TLS updates for the external CA
+type ExternalCATLSUpdater interface {
+	NotifyTLSChange() <-chan events.Event
+	GetCertificates() []tls.Certificate
+}
+
+// SecurityConfigTLSUpdater takes a security config and sets a watch so we can tell when TLS credentials have changed
+type SecurityConfigTLSUpdater struct {
```

Comment:
I was planning on removing the security config entirely from the CA server at a later point, as part of separating out the security config from the CA server, in which case I just needed some kind of interface from which I can get the updated external CA creds.

Perhaps it'd be better just to pass a channel?